nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/QuerySolr.java

```diff
@@ -0,0 +1,584 @@
+                                writer.write(recordSet);
+                                writer.flush();
+                                mimeType.append(writer.getMimeType());
+                            } catch (SchemaNotFoundException e) {
+                                throw new ProcessException("Could not parse Solr response", e);
+                            }
+                        });
+                        flowFileResponse = session.putAttribute(flowFileResponse, CoreAttributes.MIME_TYPE.key(), mimeType.toString());
+                    }
+
+                    if (processFacetsAndStats) {
+                        if (searchComponents.contains(FacetParams.FACET)) {
+                            FlowFile flowFileFacets = session.create(flowFileResponse);
+                            flowFileFacets = session.write(flowFileFacets, out -> {
+                                try (
+                                        final OutputStreamWriter osw = new OutputStreamWriter(out);
+                                        final JsonWriter writer = new JsonWriter(osw)
+                                ) {
+                                    addFacetsFromSolrResponseToJsonWriter(response, writer);
+                                }
+                            });
+                            flowFileFacets = session.putAttribute(flowFileFacets, CoreAttributes.MIME_TYPE.key(), MIME_TYPE_JSON);
+                            session.transfer(flowFileFacets, FACETS);
+                        }
+
+                        if (searchComponents.contains(StatsParams.STATS)) {
+                            FlowFile flowFileStats = session.create(flowFileResponse);
+                            flowFileStats = session.write(flowFileStats, out -> {
+                                try (
+                                        final OutputStreamWriter osw = new OutputStreamWriter(out);
+                                        final JsonWriter writer = new JsonWriter(osw)
+                                ) {
+                                    addStatsFromSolrResponseToJsonWriter(response, writer);
+                                }
+                            });
+                            flowFileStats = session.putAttribute(flowFileStats, CoreAttributes.MIME_TYPE.key(), MIME_TYPE_JSON);
+                            session.transfer(flowFileStats, STATS);
+                        }
+                        processFacetsAndStats = false;
+                    }
+                }
+
+                if (getEntireResults) {
+                    final Integer totalDocumentsReturned = solrQuery.getStart() + solrQuery.getRows();
+                    if (totalDocumentsReturned < totalNumberOfResults) {
+                        solrQuery.setStart(totalDocumentsReturned);
+                        session.transfer(flowFileResponse, RESULTS);

```

Comment:
Needs a call to the provenance manager to track the provenance. I think the RECEIVE event is the right one here.